app/assets/javascripts/admin/addon/services/admin-tools.js

```diff
@@ -39,42 +41,32 @@ export default class AdminToolsService extends Service {
     };
   }
 
-  _showControlModal(type, user, opts) {
+  @action
+  async showControlModal(type, user, opts) {
     opts = opts || {};
 
-    const controller = showModal(`admin-penalize-user`, {
-      admin: true,
-      modalClass: `${type}-user-modal`,
-    });
-
-    controller.setProperties({
-      penaltyType: type,
-      postId: opts.postId,
-      postEdit: opts.postEdit,
-    });
-
-    return (
-      user.adminUserView
-        ? Promise.resolve(user)
-        : AdminUser.find(user.get("id"))
-    ).then((loadedUser) => {
-      controller.setProperties({
+    const userLookup = user.adminUserView
+      ? Promise.resolve(user)
+      : AdminUser.find(user.get("id"));
+    const loadedUser = await userLookup;
```

Comment:
I think this should work?
```suggestion
    const loadedUser = user || await userLookup;
```